operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -127,56 +123,66 @@ private void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Con
     }
 
     private PodSpec createPodSpec(Context<Keycloak> context) {
-        var allVolumes = getAllVolumes(context);
-        Collection<String> requiredVolumes = new HashSet<>();
-        var builder = new PodSpecBuilder();
+        StatefulSet current = ContextUtils.getCurrentStatefulSet(context).orElseThrow();
+        StatefulSet desired = ContextUtils.getDesiredStatefulSet(context);
+
+        // start off with the desired statefulset state
+        var builder = desired.getSpec().getTemplate().getSpec().edit();
         builder.withRestartPolicy("Never");
-        addInitContainer(builder, context, allVolumes.keySet(), requiredVolumes);
-        addContainer(builder, context, allVolumes.keySet(), requiredVolumes);
+
+        // remove things we don't want
+
+        // TODO: this may actually not be correct - via the unsupported PodTemplate someone could make configuration dependent upon an init container
+        builder.withInitContainers();
+        builder.withContainers();
+
+        // TODO: should the scheduling fields be tweaked
```

Comment:
I'd say it would be usually safe to keep them as is. Some people would want to run Keycloak only on specific nodes, and could for example also ensure that it runs only on ARM architecture nodes. 

As an alternative, we might eventually add some separate scheduling to the upgrade part of the CR. Until then, I think we're good to keep it as is.

Reply:
> Until then, I think we're good to keep it as is.

I'm fine with that, just wanted to highlight a behavioral difference between what the logic was doing and will now do. Let's just expand the comment to explain why keeping the scheduling is ok.

Reply:
agreed, lease expand the comment.